post-merge-review.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -e
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+REGEX_REF='^\(cherry picked from commit ([a-f0-9]{40})\)$'
+
+do_reference_check() {
+  if ! [[ $(git log --format=%B "$line^!" | grep -E "$REGEX_REF") =~ $REGEX_REF ]]; then
+    echo -e "${RED}FAIL${NC}" >> "$GITHUB_OUTPUT"
+    dirty=1
+    echo "  Cannot find reference in commit message" >> "$GITHUB_OUTPUT"
+    echo "  Hint: use -x with git cherry-pick" >> "$GITHUB_OUTPUT"
+    return;
+  fi
+
+  if ! eval "$(git branch --contains "${BASH_REMATCH[1]}" "$check_branch" 2>/dev/null)"; then
```

Comment:
`eval` is almost always not needed (and evil).

`eval` is needed only when you want to compute some name and then evaluate it, imagine a "double $$". This does not seem to be the case here.

Reply:
If you want to test for an empty (resp. non-empty) string just use `test -z "$(git branch...)"` (resp. -n).

Reply:
I'm a bit confused what is being checked here: how could Github pass wrong parameters? Maybe add a comment?

Reply:
this eval came from shellcheck, updated to test.

its not checking github, its checking the `cherry picked from` line in the commit message actually exists and isn't a bogus sha